providers/cxgb4/qp.c

```diff
@@ -254,6 +270,58 @@ static int build_rdma_write(struct t4_sq *sq, union t4_wr *wqe,
 	return 0;
 }
 
+static void build_immd_cmpl(struct t4_sq *sq, struct fw_ri_immd_cmpl *immdp,
+			    struct ibv_send_wr *wr)
+{
+	memcpy((u8 *)immdp->data, (u8 *)(uintptr_t)wr->sg_list->addr, 16);
+	memset(immdp->r1, 0, 6);
+	immdp->op = FW_RI_DATA_IMMD;
+	immdp->immdlen = 16;
+}
+
+#define BUILD_BUG_ON(condition) ((void)sizeof(char[1 - 2 * !!(condition)]))
```

Comment:
Nope, use C11 static_assert for this stuff.

Reply:
Replaced BUILD_BUG_ON() with static_assert()

Reply:
Generated kernel header cpmmit on top of `94245f4ad9e10c161affaa4763136d29fbe91cb9`.